native/cocos/core/Root.cpp

```diff
@@ -120,13 +161,19 @@ void Root::destroy() {
 
     CC_SAFE_DELETE(_batcher);
 
+    for (auto *swapchain : _swapchains) {
+        CC_SAFE_DELETE(swapchain);
+    }
+    _swapchains.clear();
```

Comment:
It seems Root owns swap chains is not good. As root is only delete when exit program. Which means, swap chain will not be deleted even a system window is closed.

Reply:
Need to resolve in future.
[](https://github.com/cocos/cocos-engine/issues/12756)